cmd/govim/watcher.go

```diff
@@ -165,6 +87,39 @@ func (m *modWatcher) watch() {
 	}
 }
 
+func eventFilter(root string) func(string) bool {
+	return func(path string) bool {
+		if fi, err := os.Stat(path); err == nil && !fi.IsDir() {
+			path = filepath.Dir(path)
+		}
+
+		// TODO: cache ignored directories to reduce amount of os.Stat(...) calls
+		rel := strings.TrimPrefix(path, root)
```

Comment:
I would personally do another `strings.TrimPrefix` after this line with `os.PathSeparator` to get you the relative path. Saves the rather obscure logic below of starting from 1. Also, it's my understanding (and I would guess yours too) that the path we will be passed is clean, correct?

Reply:
Interesting, that obscure logic looks like a left-over from something else. I think that the original intent was to avoid checking for the root `go.mod` file but that won't happen anyway. I'll update.